build-system/tasks/helpers.js

```diff
@@ -453,42 +457,46 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
     };
   }
 
-  /**
-   * Bundles an entry point with all its imports and applies babel transforms.
-   * @param {boolean} continueOnError
-   */
-  async function performBundle(continueOnError) {
-    const {banner, footer} = splitWrapper();
-    await esbuild
-      .build({
-        entryPoints: [entryPoint],
-        bundle: true,
-        sourcemap: true,
-        define: experimentDefines,
-        outfile: destFile,
-        plugins: [esbuildBabelPlugin],
-        banner,
-        footer,
-      })
-      .catch((err) => handleBundleError(err, continueOnError, destFilename));
-    finishBundle(srcFilename, destDir, destFilename, options);
-    let name = destFilename;
-    if (options.latestName) {
-      const latestMaxName = options.latestName.split('.js')[0] + '.max.js';
-      name = `${name} → ${latestMaxName}`;
-    }
-    endBuildStep('Compiled', name, startTime);
-    const target = path.basename(destFilename, path.extname(destFilename));
-    if (UNMINIFIED_TARGETS.includes(target)) {
-      await applyAmpConfig(
-        destFile,
-        /* localDev */ true,
-        /* fortesting */ options.fortesting
+  const {banner, footer} = splitWrapper();
+  async function onRebuild(error, result) {
+    if (error) {
+      log.error(
+        red(`Error while recompiling ${destFilename}`),
+        `: ${error.message}`
       );
+      return;
     }
+
+    log(green(`Recompiled ${destFilename}.`));
+    if (result.warnings.length > 0) {
+      for (const warning of result.warnings) {
+        log.warn(`${yellow('Warning')} in ${warning.file}: ${warning.text}`);
+      }
+    }
+    await postBundle(srcFilename, destDir, destFilename, options);
   }
 
-  await performBundle(options.continueOnError);
+  await esbuild
+    .build({
+      entryPoints: [entryPoint],
+      bundle: true,
+      sourcemap: true,
+      define: experimentDefines,
+      outfile: destFile,
+      plugins: [esbuildBabelPlugin],
+      banner,
+      footer,
+      incremental: options.watch,
+      watch: options.watch ? {onRebuild} : false,
+      logLevel: 'silent',
+    })
+    .catch((err) =>
+      handleBundleError(err, options.continueOnError, destFilename)
+    );
+  const name = options?.latestName?.replace(/\.js$/, '.max.js') ?? destFilename;
+  endBuildStep('Compiled', `${srcFilename} → ${name}`, startTime);
```

Comment:
This line is an indicator that the rebuild is complete. I'm not seeing it when `amp.js` is rebuilt by the watcher after edit. I think everything after the call to `esbuild.build()` needs to be moved into `onRebuild`.

Reply:
This line says "Compiled". There is a separate message beginning with "Recompiled" that you should be seeing 🤔